.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,58 @@
+# This workflow runs a Criterion benchmark on a PR and compares the results against the base branch.
```

Comment:
Extra review facts:

* This is actually a rename-and-change of pr_criterion, but git has got confused
* For the branches that behave differently on push-to-main, i've tested they work by inverting the logic and pushing. I am confident the "template magic" is good.